featuremanagement/aio/_featuremanager.py

```diff
@@ -43,54 +47,65 @@ def __init__(self, configuration, **kwargs):
             self._filters[feature_filter.name] = feature_filter
 
     @staticmethod
-    def _check_default_disabled_variant(feature_flag):
+    def _check_default_disabled_variant(evaluation_event):
         """
         A method called when the feature flag is disabled, to determine what the default variant should be. If there is
         no allocation, then None is set as the value of the variant in the EvaluationEvent.
 
-        :param FeatureFlag feature_flag: Feature flag object.
-        :return: EvaluationEvent
+        :param EvaluationEvent evaluation_event: Evaluation event object.
         """
-        if not feature_flag.allocation:
-            return EvaluationEvent(enabled=False)
-        return FeatureManager._check_variant_override(
-            feature_flag.variants, feature_flag.allocation.default_when_disabled, False
+        evaluation_event.reason = VariantAssignmentReason.DEFAULT_WHEN_DISABLED
+        if not evaluation_event.feature.allocation:
+            evaluation_event.enabled = False
```

Comment:
In line 286,
``` python
if not evaluation_event.enabled:
    FeatureManager._check_default_disabled_variant(evaluation_event)
```

Setting `evaluation_event.enabled = False` in `_check_default_disabled_variant` is redundant.

In line 327,

``` python
if not feature_flag.enabled:
    # Feature flags that are disabled are always disabled
    FeatureManager._check_default_disabled_variant(evaluation_event)
    if feature_flag.allocation:
        variant_name = feature_flag.allocation.default_when_disabled
        evaluation_event.variant = self._variant_name_to_variant(feature_flag, variant_name)
    evaluation_event.feature = feature_flag

    # If a feature flag is disabled and override can't enable it
    evaluation_event.enabled = False
    return evaluation_event
```

After calling  `_check_default_disabled_variant`, you set `evaluation_event.enabled = False` again.

Reply:
For the second part, I still want to set the variant info for is_enabled as it will be provided to telemetry either way. And I believe that is how DotNet also provides things.